miner/worker_test.go

```diff
@@ -509,3 +510,214 @@ func testGetSealingWork(t *testing.T, chainConfig *params.ChainConfig, engine co
+	dbConfig := &triedb.Config{}
+	tdb := triedb.NewDatabase(db, dbConfig)
+
+	// Generate a key that will act as the Clique signer
+	testBankKey, _ := crypto.GenerateKey()
+	testBankAddress := crypto.PubkeyToAddress(testBankKey.PublicKey)
+
+	// Create the chain config (Clique)
+	config := *params.AllCliqueProtocolChanges
+	config.Clique = &params.CliqueConfig{Period: 1, Epoch: 30000}
+
+	// Instantiate the Clique engine
+	engine := clique.New(config.Clique, db)
+
+	// Authorize the signer so blocks can be sealed
+	engine.Authorize(testBankAddress, func(a accounts.Account, s string, data []byte) ([]byte, error) {
+		hash := crypto.Keccak256(data)
+		return crypto.Sign(hash, testBankKey)
+	})
+
+	// Prepare the extra data for a single signer: 32 bytes vanity + 20 bytes address + 65 bytes signature
+	extra := make([]byte, 32+common.AddressLength+crypto.SignatureLength)
+	copy(extra[32:32+common.AddressLength], testBankAddress[:])
+
+	addressBookAddress := common.HexToAddress("0x1111111111111111111111111111111111111111")
+
+	randomAddress := common.HexToAddress("0x2222222222222222222222222222222222222222")
+	randomAddressHash := common.BytesToHash(common.LeftPadBytes(randomAddress.Bytes(), 32))
+
+	// Priority address (the signer) and another address
+	priorityKey := testBankKey
+	priorityAddress := testBankAddress
+
+	priorityAddressHash := common.BytesToHash(common.LeftPadBytes(priorityAddress.Bytes(), 32))
+
+	otherKey, _ := crypto.GenerateKey()
+	otherAddress := crypto.PubkeyToAddress(otherKey.PublicKey)
+
+	// Construct the genesis with initial allocations, storage, and correct extra data
+	genesis := &core.Genesis{
+		Config:    &config,
+		Timestamp: 1,
+		ExtraData: extra,
+		Alloc: core.GenesisAlloc{
+			addressBookAddress: {
+				Storage: map[common.Hash]common.Hash{
+					common.HexToHash("0x01"): randomAddressHash,
+				},
+			},
+			randomAddress: {
+				Storage: map[common.Hash]common.Hash{
+					common.HexToHash("0x00"): priorityAddressHash,
+				},
+			},
+
+			// Both addresses need balance to pay for gas
+			otherAddress: {
+				Balance: big.NewInt(1e18),
+			},
+			priorityAddress: {
+				Balance: big.NewInt(1e18),
+			},
+		},
+	}
+
+	// Commit the genesis block
+	if _, err := genesis.Commit(db, tdb); err != nil {
+		t.Fatalf("Failed to commit genesis: %v", err)
+	}
+	// Create the blockchain from the genesis
+	blockchain, err := core.NewBlockChain(db, nil, genesis, nil, engine, vm.Config{}, nil, nil)
+	if err != nil {
+		t.Fatalf("Failed to create blockchain: %v", err)
+	}
+	defer blockchain.Stop()
+
+	// Build a backend for the worker based on this blockchain
+	backend := newCustomTestWorkerBackend(t, &config, engine, db, blockchain)
+
+	priorityConfig := &Config{
+		Recommit:                   time.Second,
+		GasCeil:                    params.GenesisGasLimit,
+		AddressBookContractAddress: &addressBookAddress,
+	}
+
+	// Create a worker
+	w := newWorker(priorityConfig, &config, engine, backend, new(event.TypeMux), nil, false)
```

Comment:
`event.TypeMux` also marked as deprecated, but the suggested replacement doesnt work seemingly without changing the newWorker signature so just leave as is